chainstate/src/detail/ban_score.rs

```diff
@@ -99,6 +99,10 @@ impl BanScore for ConnectTransactionError {
             // Even though this is an invariant, we consider it a violation to be overly cautious
             ConnectTransactionError::SerializationInvariantError(_) => 100,
             ConnectTransactionError::TimeLockViolation => 100,
+            &ConnectTransactionError::MissingBlockUndo(_) => 0,
+            &ConnectTransactionError::MissingBlockRewardUndo(_) => 0,
+            &ConnectTransactionError::MissingTxUndo(_, _) => 0,
+            &ConnectTransactionError::UtxoInvariantBroken(_) => 0,
```

Comment:
Could you please explain to me why these errors have zero ban score?

Reply:
Because it indicates something broken in the node and its invariants and not maliciousness of a peer.

Reply:
@TheQuantumPhysicist Makes sense, but we have the "Even though this is an invariant, we consider it a violation to be overly cautious" comment there and and `SerializationInvariantError` has 100 ban score. I'm just trying to understand the logic behind that.

Reply:
Look at the comments there. We can have a call some time about it if the comments aren't enough. I need to revisit them to remember the details.

The simple version is that some times we act cautiously because there could be bugs we don't anticipate that can create an attack vector. When in doubt, go for 100.